core/src/main/scala/kafka/server/KafkaHealthcheck.scala

```diff
@@ -117,14 +121,25 @@ class KafkaHealthcheck(brokerId: Int,
 
     @throws[Exception]
     override def handleNewSession() {
+      if (reconnectingMs != -1){
+        reconnectingMs = -1
+      }
       info("re-registering broker info in ZK for broker " + brokerId)
       register()
       info("done re-registering broker")
       info("Subscribing to %s path to watch for new topics".format(ZkUtils.BrokerTopicsPath))
     }
 
     override def handleSessionEstablishmentError(error: Throwable) {
-      fatal("Could not establish session with zookeeper", error)
+      this.error("Session Establishment Error: %s, waiting for %d ms before exiting".format(error, connectionRetryMs))
+      if (reconnectingMs == -1){
+        reconnectingMs = Calendar.getInstance().getTimeInMillis()
```

Comment:
why not `System.currentTimeMillis`?

Reply:
After looking further I decided this design wont work to clean up the state. Cleaned it up further and using `System.currentTimeMillis` now.